openfga/v1/authzmodel.proto

```diff
@@ -7,58 +7,61 @@ import "protoc-gen-openapiv2/options/annotations.proto";
-      example: "\"01G5JAVJ41T49E9TT3SKVS7X1J\""
-    },
-    (validate.rules).string = {
-      pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$"
-    }
+    (validate.rules).string = { pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$" },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = { example: '"01G5JAVJ41T49E9TT3SKVS7X1J"' }
   ];
-  repeated TypeDefinition type_definitions = 2 [json_name = "type_definitions"];
-}
 
-message TypeDefinitions {
-  repeated TypeDefinition type_definitions = 1 [
+  repeated TypeDefinition type_definitions = 2 [
     json_name = "type_definitions",
-
-    (validate.rules).repeated = {
-      min_items: 1
-    },
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
-      minimum: 1,
+      example: '[{"type": "user"}, {"type":"document","relations":{"reader":{"union":{"child":[{"this":{}},{"computedUserset":{"object":"","relation":"writer"}}]}},"writer":{"this":{}}},"metadata":{"relations":{"reader":{"directly_related_user_types":[{"type":"user"}]},"writer":{"directly_related_user_types":[{"type":"user"}]}}}}]'
     }
   ];
+
+  string schema_version = 3 [json_name = "schema_version"];
 }
 
 message TypeDefinition {
-  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
-    example: "{\"type\":\"document\",\"relations\":{\"reader\":{\"union\":{\"child\":[{\"this\":{}},{\"computedUserset\":{\"object\":\"\",\"relation\":\"writer\"}}]}},\"writer\":{\"this\":{}}}}"
-  };
   string type = 1 [
-    (validate.rules).string = {
-      pattern: "^[^:#@\\s]{1,254}$"
-    },
+    (validate.rules).string = { pattern: "^[^:#@\\s]{1,254}$" },
     (google.api.field_behavior) = REQUIRED,
-    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
-      example: "\"document\""
-    }
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = { example: '"document"' }
   ];
+
   map<string, Userset> relations = 2 [
-    (validate.rules).map = {
-      min_pairs: 1
-    },
-    (validate.rules).map.keys.string = {
-      pattern: "^[^:#@\\s]{1,50}$"
-    },
+    (validate.rules).map.keys.string = { pattern: "^[^:#@\\s]{1,50}$" },
     (google.api.field_behavior) = REQUIRED,
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
-      example: "{\"relations\":{\"reader\":{\"union\":{\"child\":[{\"this\":{}},{\"computedUserset\":{\"object\":\"\",\"relation\":\"writer\"}}]}},\"writer\":{\"this\":{}}}}"
+      example: '{"relations":{"reader":{"union":{"child":[{"this":{}},{"computedUserset":{"object":"","relation":"writer"}}]}},"writer":{"this":{}}}}'
     }
   ];
+
+  // A map whose keys are the name of the relation and whose value is the Metadata for that
+  // relation.
+  Metadata metadata = 3;
+}
+
+message Metadata {
+  map<string, RelationalMetadata> relations = 1;
+}
+
+message RelationalMetadata {
+  repeated RelationReference directly_related_user_types = 1 [ json_name = "directly_related_user_types" ];
```

Comment:
Will this be updated to be either a type or a relation reference where relation reference does not have relation as optional?

Reply:
No